### DISCUSSION TITLE
Replacement for `getStringContent` method?

### DISCUSSION THREAD

**Discussion by gagahpangeran** [thephpleague/commonmark#714] [Q&A] [answered]
Hello!

Currently I'm in the middle of upgrading commonmark from v1.6 to v2.0 for [osu-web](https://github.com/ppy/osu-web/) project.

I notice in our codebase, there's this line of code `$this->node->getStringContent()` where `$this-node` itself is `Heading` block.

Previously in v1.6, `Heading` block extends `AbstractStringContainerBlock` so it has `getStringContent()` method.

In v2.0, `AbstractStringContainerBlock` is removed and now `Heading` block extends `AbstractBlock`, so `getStringContent()` method is gone.

I already read the documentation and also dive into commonmark codebase itself but can't find any replacement for that method. Am I missing something?

**Comment by colinodell** [comment, marked as answer]
In 2.0 we no longer store the text contents directly on the parent `Heading` block node but rather inside of its children nodes.  The easiest way to extract that text in 2.0 would be with the `League\CommonMark\Node\StringContainerHelper` class like this:

```php

use League\CommonMark\Node\StringContainerHelper;

// ...

$text = StringContainerHelper::getChildText($this->node);
```

**Reply by gagahpangeran** [reply]
It works!

Thank you!
